### DISCUSSION TITLE
Formatting MDX 2 files

### DISCUSSION THREAD

**Discussion by PupoSDC** [mdx-js/mdx#2051] [Q&A] [answered]
Hello All!

Im starting to write a quite big project using MDX 2 (it's the future!). However... one thing that is being an absolute pain in the ass is formatting. Prettier MDX support is very poor and I'm looking for alternatives. What stuff are you guys using to keep your mdx files looking consistent?

**Comment by PupoSDC** [comment]
This is the core issue im facing with prettier:

https://github.com/prettier/prettier/issues/12209#issuecomment-1035005697

Basically I have quite a lof of normal MDX inside jsx, and i have to make a hard decision, either i choose to "preserve format" and prettier pretty much doesn't fix anything relevant on the mdx file, or I choose to not preserve format and I have the issue described above.

**Comment by wooorm** [comment, marked as answer]
I’m using `remark` with `remark-mdx` and a bunch of other plugins. There’s also `eslint-mdx` that can help.

https://github.com/mdx-js/mdx/blob/63fd208d53f13e5b2451c9f9bc1b7fd136d22c71/package.json#L133

https://github.com/mdx-js/mdx/blob/63fd208d53f13e5b2451c9f9bc1b7fd136d22c71/package.json#L235-L247

**Reply by PupoSDC** [reply]
Thank you very much for your answer!

After a bit of fiddling i can see this working for me, at least until prettier catches up with mdx. 

I have a question: I quite like most of your preset, but some stuff I would like to disable. For example, the smart quotes seems to not be auto fixable (?) so i would like to remove them... What would be the best way to achieve this? Create my own personnal format package like you did yours?

**Reply by wooorm** [reply]
It’s a lot of sometimes annoying messages, but, I am personally very adamant on proper quotes, so you might be able to learn to love that rule :)

Depends how much you’d disable. The above example turns of `remark-lint-no-html` and `remark-lint-file-extension`.
The natural language checking here is complex, I think it can work by passing `remark-retext` and `false`. If that doesn’t work, you’re out of luck and need your own preset!

**Comment by kachkaev** [comment]
If anyone one wants to help with MDX 2 support in Prettier, please see https://github.com/mdx-js/mdx/discussions/2127
